shared/processing/dataGetters.ts

```diff
@@ -0,0 +1,319 @@
+import {
+    ContentItem,
+    Fields,
+} from 'kentico-cloud-delivery';
+import { ZapiCategory } from '../models/zapi__category';
+import { ZapiContact } from '../models/zapi_contact';
+import { ZapiLicense } from '../models/zapi_license';
+import { ZapiParameter } from '../models/zapi_parameter';
+import { ZapiPathOperation } from '../models/zapi_path_operation';
+import { ZapiRequestBody } from '../models/zapi_request_body';
+import { ZapiResponse } from '../models/zapi_response';
+import { ZapiSecurityScheme } from '../models/zapi_security_scheme';
+import { ZapiServer } from '../models/zapi_server';
+import { ZapiSpecification } from '../models/zapi_specification';
+import {
+    ICategory,
+    IContact,
+    ILicense,
+    IParameter,
+    IPathOperation,
+    IPreprocessedData,
+    IRequestBody,
+    IResponse,
+    ISchemas,
+    ISecurityScheme,
+    IServer,
+    IWrappedItem,
+    IZapiSpecification,
+} from '../types/dataModels';
+import {
+    getFromLinkedItems,
+    processLinkedItemsElement,
+    processMultipleChoiceElement,
+    processTaxonomyElement,
+} from '../utils/processElements';
+import {
+    getSystemProperties,
+    getWrappedData,
+    processItemsInto,
+} from './common';
+import {
+    getDescriptionItemsData,
+    IDescriptionComponents,
+} from './descriptionItemGetters';
+import {
+    getSchemaDataFromLinkedItemElement,
+    getSchemaDataFromRichTextElement,
+} from './schemaGetters';
+import RichTextField = Fields.RichTextField;
+
+export const getApiSpecificationData = (
+    item: ZapiSpecification,
+    dataBlob: IPreprocessedData,
+    linkedItems: ContentItem[],
+): Array<IWrappedItem<IZapiSpecification>> => {
+    processItemsInto<ICategory>(getCategoriesData)(item.categories, dataBlob, linkedItems);
+    processItemsInto<IContact>(getContactData)(item.contact, dataBlob, linkedItems);
+    processItemsInto<ILicense>(getLicenseData)(item.license, dataBlob, linkedItems);
+    processItemsInto<IPathOperation>(getPathOperationsData)(item.pathOperations, dataBlob, linkedItems);
+    processItemsInto<ISecurityScheme>(getSecuritySchemeData)(item.security, dataBlob);
+    processItemsInto<IServer>(getServersData)(item.servers, dataBlob, linkedItems);
+    processItemsInto<IDescriptionComponents>(getDescriptionItemsData)(item.description, dataBlob, linkedItems);
```

Comment:
You recreate "first" `processItemsInto` on every `getApiSpecificationData` call - that kind of beats its purpose. Create and name these method within module itself (e.g. `const processDescriptionComponents = processItemsInto<IDescriptionComponents>(getDescriptionItemsData);`) and call these factored function within exported functions later on:
```suggestion
    processDescriptionComponents(item.description, dataBlob, linkedItems);
```